nuxeo-compound-documents/src/main/java/org/nuxeo/compound/documents/CompoundDocumentServiceImpl.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * (C) Copyright 2022 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package org.nuxeo.compound.documents;
+
+import static org.nuxeo.compound.documents.CompoundDocumentConstants.COMPOUND_DOCTYPE;
+import static org.nuxeo.compound.documents.CompoundDocumentConstants.COMPOUND_FOLDER_DOCTYPE;
+import static org.nuxeo.runtime.model.Descriptor.UNIQUE_DESCRIPTOR_ID;
+
+import java.io.IOException;
+import java.util.Comparator;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipFile;
+
+import org.apache.commons.io.FilenameUtils;
+import org.nuxeo.common.utils.Path;
+import org.nuxeo.ecm.core.api.Blob;
+import org.nuxeo.ecm.core.api.CoreSession;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.core.api.NuxeoException;
+import org.nuxeo.ecm.core.api.impl.blob.ZipEntryBlob;
+import org.nuxeo.ecm.platform.filemanager.api.FileImporterContext;
+import org.nuxeo.ecm.platform.filemanager.api.FileManager;
+import org.nuxeo.runtime.api.Framework;
+import org.nuxeo.runtime.model.DefaultComponent;
+
+/** @since 2021.0 */
+public class CompoundDocumentServiceImpl extends DefaultComponent implements CompoundDocumentService {
+
+    protected static final String XP_IGNORED_FILES = "ignoredFiles";
+
+    @Override
+    public DocumentModel createCompoundDocument(CoreSession session, String parent, Blob archive) {
+        String compoundDocName = FilenameUtils.removeExtension(archive.getFilename());
+        DocumentModel compoundDoc = session.createDocumentModel(parent, compoundDocName, COMPOUND_DOCTYPE);
+        compoundDoc = session.createDocument(compoundDoc);
+        createCompoundDocument(compoundDoc, archive);
+        return compoundDoc;
+    }
+
+    protected void createCompoundDocument(DocumentModel compoundDoc, Blob archive) {
+        try (ZipFile zip = new ZipFile(archive.getFile())) {
+            FileManager fileManager = Framework.getService(FileManager.class);
+            zip.stream().filter(this::isAllowedEntry).sorted(Comparator.comparing(ZipEntry::getName)).forEach(entry -> {
+                Path entryPath = new Path(entry.getName());
+                String parentPath = compoundDoc.getPath().append(entryPath.removeLastSegments(1)).toString();
```

Comment:
Maybe something like:
```suggestion
                Path entryPath = new Path(entry.getName());
                String entryName = entryPath.lastSegment();
                String parentPath = compoundDoc.getPath().append(entryName).toString();
```

Reply:
Yes for the local variables but you still need the ou need to call`entryPath.removeLastSegments(1)`